dbos/queue.py

```diff
@@ -0,0 +1,40 @@
+import threading
+import time
+from typing import TYPE_CHECKING, Optional
+
+from dbos.core import P, R, _execute_workflow_id, _start_workflow
+from dbos.error import DBOSInitializationError
+
+if TYPE_CHECKING:
+    from dbos.dbos import DBOS, Workflow, WorkflowHandle
+
+
+class Queue:
+    def __init__(self, name: str, concurrency: Optional[int] = None) -> None:
+        self.name = name
+        self.concurrency = concurrency
+        from dbos.dbos import _get_or_create_dbos_registry
+
+        registry = _get_or_create_dbos_registry()
+        if self.name in registry.queue_info_map:
+            raise DBOSInitializationError(
+                f"Failed to register queue {self.name}: a queue of this name already exists."
+            )
+        registry.queue_info_map[self.name] = self
+
+    def enqueue(
```

Comment:
I would like to see the queue function available as `DBOS.enqueue` with the queue name parameter.  It is otherwise a challenge to access the queue from multiple files, and not advised to make different Queue objects with the same name to fix it...

Reply:
I tried putting queues on the main DBOS object and found it much harder to work with. The main problem is that if you do it that way, you have to type in the queue name as a string literal in multiple places, because if you're passing it around as a variable you might as well use the queue object.

Also, in most popular queuing libraries (like BullMQ), a queue is an object. More intuitive interface in my opinion.